components/com_users/tmpl/login/default_login.php

```diff
@@ -35,7 +35,7 @@
     </div>
     <?php endif; ?>
 
-    <?php if (($this->params->get('logindescription_show') == 1 && str_replace(' ', '', $this->params->get('login_description', '')) != '') || $this->params->get('login_image') != '') : ?>
+    <?php if (($this->params->get('logindescription_show') == 1 && trim($this->params->get('login_description', ''))) || $this->params->get('login_image') != '') : ?>
```

Comment:
```suggestion
    <?php if (($this->params->get('logindescription_show') == 1 && trim($this->params->get('login_description', ''), ' ') !== '') || $this->params->get('login_image') != '') : ?>
```
I think we should do it this way.

Reply:
Trim is still faster than str_replace because it doesn't have to evaluate the whole string in all cases.

Reply:
As I wrote earlier, I disagree. We want to remove all invisible characters and not just spaces.